File: ClassroomMgmt-Magazine/src/models/Training.ts
import { Station } from "./Station"; 

export interface Training {
  id: string;
  location: string | null;
  title: string;
  isLive: boolean;
  courseId?: string | null;
  instructorId?: string | null;
  stationIds?: string[] | null;
  stations?: Station[] | null;
  trainingStatisticsId?: string | null;
  studyUnitId: string;
  creationDate: Date;
  version: number;
}

export const NEW_TRAINING_TEMPLATE: Training = {
  id: "", 
  location: "",
  title: "",
  isLive: false,
  courseId: null,
  instructorId: null,
  stationIds: [],
  stations: [],
  trainingStatisticsId: null,
  studyUnitId: "",
  creationDate: new Date(),
  version: 1
};

File: ClassroomServer/Models/Training.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ClassroomServer.Models
{
    public class Training
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        public string? Location { get; set; } = null!;

        [Required]
        public string Title { get; set; } = null!;

        public bool IsLive { get; set; }

        public Guid? CourseId { get; set; }

        public Guid? InstructorId { get; set; }

        public List<Guid>? StationIds { get; set; } = null;

        public List<Station>? Stations { get; set; } = null;

        public Guid? TrainingStatisticsId { get; set; } = null;

        public virtual TrainingStatistics? TrainingStatistics { get; set; } = null;

        public Guid StudyUnitId { get; set; }

        public DateTime CreationDate { get; set; }

        public int Version { get; set; }
    }
}
